app/src/components/keyboardNavigation.spec.js

```diff
@@ -0,0 +1,148 @@
+      "curSlideIndex"
+    );
+    spyMaybeGoToNextSlide = jest.spyOn(
+      KeyboardNavigation.prototype,
+      "maybeGoToNextSlide"
+    );
+    spyMaybeGoToPrevSlide = jest.spyOn(
+      KeyboardNavigation.prototype,
+      "maybeGoToPrevSlide"
+    );
+    spyHandleKeydown = jest.spyOn(
+      KeyboardNavigation.prototype,
+      "handleKeyDown"
+    );
+    spyRemoveEvents = jest.spyOn(KeyboardNavigation.prototype, "removeEvents");
+    setDocumentHtml(getMainHtml()); // eslint-disable-line no-undef
+  });
+
+  beforeEach(() => {
+    keyboardNavigation = new KeyboardNavigation({
+      element: document.body,
+      store,
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    jest.resetModules();
+  });
+
+  test("The consumer should be able to call new() on KeyboardNavigation", () => {
+    expect(keyboardNavigation).toBeTruthy();
+  });
+
+  test("Throws an error if props.store is missing or invalid", () => {
+    expect(
+      () =>
+        new KeyboardNavigation({
+          element: document.body,
+        })
+    ).toThrow("Requires redux store");
+
+    expect(
+      () =>
+        new KeyboardNavigation({
+          element: document.body,
+          store: {},
+        })
+    ).toThrow("Requires redux store");
+  });
+
+  test("curSlideIndex returns the value of slides.curIndex from the redux store", () => {
+    keyboardNavigation.curSlideIndex();
+    expect(store.getState).toHaveBeenCalled();
+    expect(spyCurSlideIndex).toHaveReturnedWith(0);
+  });
+
+  test("maybeGoToNextSlide should dispatch GoToNextSlide action if permissible", () => {
+    keyboardNavigation.maybeGoToNextSlide();
+    expect(store.dispatch).toHaveBeenCalledWith({ type: "GoToNextSlide" });
+  });
+
+  test("maybeGoToNextSlide should not call store.dispatch if not permissible", () => {
+    store.getState.mockImplementationOnce(() => ({
+      slides: {
+        curIndex: 200,
+      },
+    }));
+    keyboardNavigation.maybeGoToNextSlide();
+    expect(store.dispatch).not.toHaveBeenCalled();
+  });
+
+  test("maybeGoToPrevSlide should dispatch GoToPrevSlide action if permissible", () => {
+    store.getState.mockImplementationOnce(() => ({
+      slides: {
+        curIndex: 1,
+      },
+    }));
+    keyboardNavigation.maybeGoToPrevSlide();
+    expect(store.dispatch).toHaveBeenCalledWith({ type: "GoToPrevSlide" });
+  });
+
+  test("maybeGoToPrevSlide should not call store.dispatch if not permissible", () => {
+    store.getState.mockImplementationOnce(() => ({
+      slides: {
+        curIndex: 0,
+      },
+    }));
+    keyboardNavigation.maybeGoToPrevSlide();
+    expect(store.dispatch).not.toHaveBeenCalled();
+  });
+
+  test("handleKeyDown is called on document keydown event", () => {
+    document.dispatchEvent(new Event("keydown"));
+    expect(spyHandleKeydown).toHaveBeenCalled();
+  });
+
+  test("handleKeyDown calls class methods when receiving correct key code", () => {
```

Comment:
add a test to make sure `handleKeyDown` does not call class methods when `event.target` is not the `body` element.